exercises/connect/example.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * "Player O" plays from top to bottom, "Player X" plays from left to right.
+ * @param {string[]} initBoard - The starting state of the board
+ * @returns {Object} - The board business object
+ */
+var Board = function (initBoard) {
+  this.board = initBoard.slice();
+};
+
+Board.prototype.winner = function () {
+  var players = ['X', 'O'];
+  for (var i = 0; i < players.length; i++) {
+    if (this.checkWin(players[i])) {
+      return players[i];
+    }
+  }
+  return '';
+};
+
+Board.prototype.checkWin = function (player) {
+  var positions = this.startPositions(player);
+  for (var i = 0; i < positions.length; i++) {
+    if (this.search(positions[i], player, [])) {
+      return true;
+    }
+  }
+  return false;
+};
+
+Board.prototype.search = function (pos, XorO, isChecked) {
+  var boardInstance = this;
+  if (!this.matches(pos, XorO)) {
+    return false;
+  }
+  if (this.winningSpot(pos, XorO)) {
+    return true;
+  }
+  var checked = isChecked.slice(0);
+  checked.push(pos);
+
+  var matches = this.neighbors(pos).filter(function (cell) {
+    return boardInstance.matches(cell, XorO) &&
+      checked.filter(function (spot) { return spot.x === cell.x && spot.y === cell.y; }).length === 0;
+  });
+
+  if (matches.length === 0) {
+    return false;
+  }
+
+  return matches.filter(function (spot) {return boardInstance.search(spot, XorO, checked);}).length > 0;
```

Comment:
I feel this line is a bit long.

An alternative could be to extract a local variable, but I'm not sure it would make the code clearer. What's your opinion?

Reply:
I agree that it's a bit long, and we also have a local `matches` and a method `matches`.  I think I should change that.